client/state/selectors/is-requesting-media.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import MediaQueryManager from 'lib/query-manager/media';
+
+/**
+ * Returns true if media is being requested for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {bool}           True if media is being requested
+ */
+export default function isRequestingMedia( state, siteId, query ) {
+	const queryRequests = state.media.queryRequests[ siteId ];
+	const stringifiedQuery = MediaQueryManager.QueryKey.stringify( query );
+
+	if ( ! queryRequests ) {
```

Comment:
Since we've not yet used `stringifiedQuery` at this point where an early return could occur, and `QueryKey.stringify` is not exactly trivial, it might be better to declare after the condition.